farmerbot/internal/farmerbot.go

```diff
@@ -0,0 +1,425 @@
+
+		if err := json.Unmarshal(payload, &options); err != nil {
+			return nil, fmt.Errorf("failed to load request payload: %w", err)
+		}
+
+		nodeID, err := f.findNode(subConn, options)
+		return nodeID, err
+	})
+
+	powerRouter.WithHandler("includenode", func(ctx context.Context, payload []byte) (interface{}, error) {
+		err := authorize(ctx, farmerTwinID)
+		if err != nil {
+			return nil, err
+		}
+
+		var nodeID uint32
+		if err := json.Unmarshal(payload, &nodeID); err != nil {
+			return nil, fmt.Errorf("failed to load request payload: %w", err)
+		}
+
+		_, ok := f.nodes[nodeID]
+		if ok {
+			return nil, fmt.Errorf("node %d already exists", nodeID)
+		}
+
+		if slices.Contains(f.config.ExcludedNodes, nodeID) ||
+			len(f.config.ExcludedNodes) == 0 && !slices.Contains(f.config.IncludedNodes, nodeID) {
+			return nil, fmt.Errorf("node %d is excluded, cannot add it", nodeID)
+		}
+
+		neverShutDown := slices.Contains(f.config.NeverShutDownNodes, nodeID)
+		node, err := getNode(ctx, subConn, f.rmbNodeClient, nodeID, neverShutDown, false, f.farm.DedicatedFarm, on)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to include node with id %d", nodeID)
+		}
+
+		f.state.addNode(node)
+		return nil, nil
+	})
+
+	powerRouter.WithHandler("poweroff", func(ctx context.Context, payload []byte) (interface{}, error) {
+		err := authorize(ctx, farmerTwinID)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := f.validateAccountEnoughBalance(subConn); err != nil {
+			return nil, fmt.Errorf("failed to validate account balance: %w", err)
+		}
+
+		var nodeID uint32
+		if err := json.Unmarshal(payload, &nodeID); err != nil {
+			return nil, fmt.Errorf("failed to load request payload: %w", err)
+		}
+
+		if err := f.powerOff(subConn, nodeID); err != nil {
+			return nil, fmt.Errorf("failed to power off node %d: %w", nodeID, err)
+		}
+
+		// Exclude node from farmerbot management
+		// (It is not allowed if we tried to power on a node the farmer decided to power off)
+		// the farmer should include it again if he wants to the bot to manage it
+		f.state.deleteNode(nodeID)
+		return nil, nil
+	})
+
+	powerRouter.WithHandler("poweron", func(ctx context.Context, payload []byte) (interface{}, error) {
+		err := authorize(ctx, farmerTwinID)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := f.validateAccountEnoughBalance(subConn); err != nil {
+			return nil, fmt.Errorf("failed to validate account balance: %w", err)
+		}
+
+		var nodeID uint32
+		if err := json.Unmarshal(payload, &nodeID); err != nil {
+			return nil, fmt.Errorf("failed to load request payload: %w", err)
+		}
+
+		if err := f.powerOn(subConn, nodeID); err != nil {
+			return nil, fmt.Errorf("failed to power on node %d: %w", nodeID, err)
+		}
+
+		// Exclude node from farmerbot management
+		// (It is not allowed if we tried to power off a node the farmer decided to power on)
+		// the farmer should include it again if he wants to the bot to manage it
```

Comment:
So at this point this node is excluded even if the farmer defined it in their config?

Reply:
It is mentioned in the README